### PR TITLE
Feature/set restriction

### PR DESCRIPTION
**Pull request by TobiasMeissner2000**

### CODE REVIEW

File: CfStreamUploader/CfStreamUploader.Core/VideoManager.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                {"kid", config.UserSettings.KeyId},
+                {"typ", "JWT"}
+            };
+            var payload = new Dictionary<string, object>()
+            {
+                {"sub", videoId},
+                {"kid", config.UserSettings.KeyId},
+                {"exp", this.GetExpireDate(checkboxRestrictionExpireIn, config)},
+                {
+                    "accessRules", this.AccesRulesManager(config, checkboxRestrictionIp, checkboxRestrictionCountry,
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Core/VideoManager.cs:51]: GetAccessRules

File: CfStreamUploader/CfStreamUploader.Presentation/Resources/Styles/LoadingAnimationStyle.xaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
+﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/Resources/Styles/LoadingAnimationStyle.xaml:1]: DELETE

File: CfStreamUploader/CfStreamUploader.Presentation/Resources/Styles/TextBlockStyles.xaml [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
     <Style TargetType="TextBlock" x:Key="UploadProgressTextBlockStyle">
         <Setter Property="Margin" Value="5" />
-        <Setter Property="Foreground" Value="{Binding ProgressColor}" />
+        <!-- <Setter Property="Foreground" Value="{Binding ProgressColor}" /> -->
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/Resources/Styles/TextBlockStyles.xaml:20]: ?

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditRestrictionViewModel.cs [hunk 4]
```diff
@@ -170,13 +180,23 @@ private void SaveButton()
             var countryString = this.CountryTextBox.Replace(" ", "");
             this.ConfigManager.Config.AccessRules.Country.SetCountryList(countryString.Split(",").ToList());
 
+            if (!this.ExpiresInTextBox.All(char.IsDigit))
+            {
+                MessageBox.Show("Make shure your AccesLimit is a Number", "Warning", MessageBoxButton.OK,
```
Comment (TobiasMeissner2000) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditRestrictionViewModel.cs:185]: acceslimit
number

Comment (TobiasMeissner2000) [reply CfStreamUploader/CfStreamUploader.Presentation/ViewModels/EditRestrictionViewModel.cs:185]: Readme.md --> Config

File: CfStreamUploader/CfStreamUploader.Core/VideoManager.cs [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                {
+                    JsonConvert.DeserializeObject(jsonStringList[0]),
+                    JsonConvert.DeserializeObject(jsonStringList[1]),
+                    JsonConvert.DeserializeObject(jsonStringList[2])
+                },
+            };
+
+            return accesruleJson;
+        }
+
```
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Core/VideoManager.cs:149]: Switch Methods

File: CfStreamUploader/CfStreamUploader.Presentation.Test/TestSamples/Samples.cs [hunk 6]
```diff
@@ -25,14 +17,13 @@ public static class Samples
             {
                 AccessRules = new AccessRules()
                 {
-                    Any = { Action = "allow" },
-                    Ip = { Action = "allow", Ips = { "TestIp" } },
-                    Country = { Action = "allow", Countries = { "DE", "FR" } }
+                    Any = {Action = "allow"},
+                    Ip = {Action = "allow", Ips = {"TestIp"}},
+                    Country = {Action = "allow", Countries = {"DE", "FR"}},
+                    ExpiresIn = 10
```
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation.Test/TestSamples/Samples.cs:23]: 365 ?

File: CfStreamUploader/CfStreamUploader.Presentation/Converter/BooleanToColorConverter.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+{
+    public class BooleanToColorConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if ((bool)value)
+            {
+                return "Green";
+            }
+            return "#6497e8";
```
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation/Converter/BooleanToColorConverter.cs:16]: Write the color in a comment

File: CfStreamUploader/CfStreamUploader.Presentation/Resources/Styles/TextBlockStyles.xaml (hunk 3 above)
Comment (LeoEcker) [reply CfStreamUploader/CfStreamUploader.Presentation/Resources/Styles/TextBlockStyles.xaml:20]: DELETE

File: CfStreamUploader/CfStreamUploader.Presentation/ViewModels/MainViewModel.cs [hunk 8]
```diff
@@ -27,9 +28,32 @@ public class MainViewModel : ViewModelBase, IDropTarget
...
-            $@"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\CfStreamUploader\Config.json";
+        private string restrictionExpireIn = string.Empty;
+        private bool checkboxRestrictionIP = false;
+        private bool checkboxRestrictionCountry = false;
+        private bool checkboxRestrictionAny = true;
+        private bool checkboxRestrictionExpireIn = true;
+        private bool setSignedUrl = true;
+
+        //VideoPogressBar
+        private bool checkImage1IsVisible;
```
Comment (LeoEcker) [inline CfStreamUploader/CfStreamUploader.Presentation/ViewModels/MainViewModel.cs:39]: checkImageOneIsVisible

**Review by LeoEcker — APPROVED** [review]

**Pull request merged**